src/main/java/org/veupathdb/service/eda/merge/core/stream/RootStreamingEntityNode.java

```diff
@@ -139,11 +140,14 @@ public Map<String,String> next() {
     _computeStreamProcessor.ifPresent(computeStream -> applyCompute(row, computeStream));
 
     // return only requested vars and in the correct order
-    _outputRow.clear();
+    final InitialSizeStringMap outputRow = _outputRowBuilder.build();
```

Comment:
It should be very cheap to build. All it stores is a reference to the key index and I thought it might be a little cleaner to avoid sharing the same mutable reference everywhere.